### PR TITLE
Catch unrecognized filter

### PR DESCRIPTION
**Pull request by esoergel**
https://manage.dimagi.com/default.asp?282382
Show a message to the user if they reference an unrecognized filter by manually editing a report builder report.

### CODE REVIEW

**Comment by esoergel** [comment]
Another thought I had here was that maybe when this error is raised we should make the report no longer a report builder report, so it automatically goes to the advanced view.  Thoughts, @kaapstorm ?

**Review by nickpell — APPROVED** [review]

File: corehq/apps/userreports/reports/builder/forms.py [hunk 1]
```diff
@@ -1011,10 +1011,17 @@ def _get_view_model(self, filter):
                 'date': 'Date',
                 'numeric': 'Numeric'
             }
+            try:
+                format_ = filter_type_map[filter['type']]
```
[the remark this answers is not in the document]
Comment (esoergel) [reply corehq/apps/userreports/reports/builder/forms.py]: Good point, though `filter['type']` has already been called at this point, actually (line 996), so I think it's okay.

[the remark this answers is not in the document]
Comment (nickpell) [reply corehq/apps/userreports/reports/builder/forms.py]: 👍

**Comment by kaapstorm** [comment]
> we should make the report no longer a report builder report, so it automatically goes to the advanced view.

I agree, @esoergel. If someone has manually edited a report, firstly, we know they are comfortable with the advanced view, and secondly, I don't think they have an expectation that the report is still a "Report Builder" report.

**Review by kaapstorm — APPROVED** [review]

**Review by gbova — APPROVED** [review]

**Comment by esoergel** [comment]
@kaapstorm  IMO it'd be nice to let them continue in report builder if the report is still compatibile, though I'm not sure how difficult that'd be to ascertain.  For instance, I could see someone dropping into UCR to copy/paste something a bunch of times.  I'm gonna table any further work on this, but it'll be good to have in mind for the next time something like this comes up.

**Pull request merged**